client/builds.go

```diff
@@ -70,6 +70,10 @@ func (c *Client) CreateBuildIndex(app string, index Index, cache bool, manifest
 }
 
 func (c *Client) CreateBuildSource(app string, source []byte, cache bool, manifest string, description string) (*Build, error) {
+	return c.CreateBuildSourceP(app, source, cache, manifest, description, nil)
+}
+
+func (c *Client) CreateBuildSourceP(app string, source []byte, cache bool, manifest string, description string, progressCallback func(s string)) (*Build, error) {
```

Comment:
let's use a slightly more descriptive name as well if we're going to export this